### PR TITLE
refs #10003 - use new config migration script [deb]

### PR DESCRIPTION
**Pull request by domcleal**
I've only been able to test this by testing the scripts standalone, which is pretty much copied from #664, as I don't have a quick way to build a package from both the smart proxy + packaging branches.

### CODE REVIEW

File: debian/jessie/foreman-proxy/postinst [hunk 1]
```diff
@@ -41,22 +41,19 @@ case "$1" in
...
-      cd /tmp && rm -rf /var/lib/foreman-proxy
-      # If we migrated, we also need to fix the settings.d dir
-      if grep -q ^:settings_directory: $CONFIGDIR/settings.yml; then
-        sed -i "s|:settings_directory:.*|:settings_directory: $CONFIGDIR/settings.d|" $CONFIGDIR/settings.yml
-      else
-        echo ":settings_directory: $CONFIGDIR/settings.d" >> $CONFIGDIR/settings.yml
-      fi
+    TEMP=$(mktemp -d)
+    trap "rm -rf $TEMP" EXIT
+    pushd $TEMP >/dev/null
```
Comment (mmoll) [inline debian/jessie/foreman-proxy/postinst]: that's a bashism:
```
/var/lib/dpkg/info/foreman-proxy.postinst: 46: /var/lib/dpkg/info/foreman-proxy.postinst: pushd: not found
```

Comment (domcleal) [inline debian/jessie/foreman-proxy/postinst]: Thanks, switched to a subshell and `cd`.

**Comment by domcleal** [comment]
Once it builds I'll also give it a bit of testing before merging.

**Comment by domcleal** [comment]
Also fixed the missing extra/migrations/ directory and it seems to be working OK now.

**Comment by mmoll** [comment]
Looking good, at least for me on Debian/jessie, thanks @domcleal!

**Pull request merged**